yt/data_objects/static_output.py

```diff
@@ -323,6 +323,14 @@ def unique_identifier(self, value):
     def periodicity(self):
         if self._force_periodicity:
             return (True, True, True)
+        elif hasattr(self, "_domain_override"):
+            # dataset loaded with a bounding box
+            if any(self._periodicity):
+                mylog.warning(
```

Comment:
This warning might trigger at surprising times because the periodicity property is often accessed internally. It would probably be preferable to issue the warning when the bounding box is created instead.